cql3/restrictions/statement_restrictions.cc

```diff
@@ -969,5 +975,229 @@ bool single_column_restriction::LIKE::is_satisfied_by(bytes_view data, const que
     return (*_matcher)(data);
 }
 
+namespace wip {
+
+namespace {
+
+using cql3::selection::selection;
+
+/// Returns cdef's value from the fetched data.
+bytes_opt get_value(const column_definition* cdef, const selection& selection,
```

Comment:
Maybe a bytes_view_opt?

Reply:
This returns serialization results, [which are prvalues](https://github.com/scylladb/scylla/blob/master/types.cc#L2729).  Wouldn't such objects be destroyed immediately, leaving bytes_view dangling?

Reply:
It looks like it returns lvalues, referring to the input. In all cases it is the result of vector::operator[], which is an lvalue.

Reply:
Oh, you're absolutely right.  Sorry, I was looking at some other code when I wrote the above.  I will make this `bytes_view_opt`.

Reply:
I figured out what I was looking at: this doesn't, in fact, always return a vector element.  When looking at a subscripted column, it returns the serialized map value, which is ephemeral, per the link I provided.